composer/loggers/progress_bar_logger.py

```diff
@@ -189,7 +189,19 @@ def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]) -> N
         # log to console
         if self.should_log(state, log_level):
             data_str = format_log_data_value(data)
-            log_str = f'[{log_level.name}][batch={int(state.timestamp.batch)}]: {data_str}'
+            assert state.max_duration is not None, 'max_duration should be set'
+
+            if state.max_duration.unit == TimeUnit.EPOCH:
+                assert state.dataloader_len is not None
```

Comment:
Can we be graceful to dataloaders of unknown length?

Reply:
I think this may be OK though because if `max_duration` is in epochs, the dataloader is required to be sized right? I think when I use infinite-sized IterableDatasets, I would expect Trainer to forbid me from using `max_duration` in units of epochs.